packages/xod-project/src/messages.js

```diff
@@ -64,6 +64,78 @@ export default {
       conflictingSpecializations
     )}`,
   }),
+  SPECIALIZATION_PATCH_CANT_BE_ABSTRACT: ({ trace }) => ({
+    title: 'Invalid specialization patch',
+    note: "Specialization patches can't be abstract",
+    trace,
+  }),
+  SPECIALIZATION_PATCH_MUST_HAVE_SAME_ARITY_LEVEL: ({ trace }) => ({
+    title: 'Invalid specialization patch',
+    note: 'Specialization patches must have same arity level as abstract',
```

Comment:
Perhaps you mean “arity step,” not “arity level?”

Nevertheless, more details are necessary:

> Specialization patches must have same arity step as abstract. However, `@/foo(number)` has step 1, whereas `@/foo` has 2.

solution:

> Change `variadic-1` marker to `variadic-2` on specialization to continue.